### PR TITLE
Graphql testing

### PR DESCRIPTION
**Pull request by kevingloss**
# Description

Please check the type of change your PR introduces:
- [ ] Bugfix (change which fixes an issue)
- [ ] Feature (change which adds functionality)
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [x] Other (please describe): 

 
Code implements further testing for all graphql endpoints.
 

### Type of change

Please check the relevant option.

- [x] Non-Breaking Change
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Big Breaking change (this breaks everything and I am sorry)

# Where should reviewers start?

Reviewers can dive right into all the testing spec files for graphQL.

# How Has This Been Tested?

Please check the option(s) that best describe your tests. 

- [x] New spec files were added
- [x] Tests were added to existing spec files
- [ ] Some tests were changed
- [x] No tests were changed
- [ ] All tests were changed (please explain what in God's name happened)

### Tests Description

- Implemented testing for all story queries and the remaining user mutations.

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have removed all unused code or commented why I left it
- [ ] I have deployed my changes to a test-heroku and they work
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

**Review by dylan-harper — APPROVED** [review]
Worked on this with Kevin and I have no suggested edits

File: spec/graphql/mutations/stories/create_story_spec.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                word
+                image
+                sound
+                totalTimeInSeconds
+                }
+              }
+            }
+          GQL
+        end
+
```
Comment (jbreit88) [inline spec/graphql/mutations/stories/create_story_spec.rb:34]: Oh, this is super cool! Normally when I pass in data like this I'll set it equal to a variable rather than a method. Does it need to be a method for graphQL, or was this just the way you thought to do it? I see nothing wrong with it, just curious if there is added benefit or detractions.

File: spec/graphql/mutations/stories/delete_story_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+              title
+            }
+          }
+        }
+        GQL
+      end
+
+      it 'deletes a story' do
+        stories = create_list(:story, 5)
+        @story = create(:story, title: 'Deleted')
```
Comment (jbreit88) [inline spec/graphql/mutations/stories/delete_story_spec.rb:44]: Not sure why we have a global @story variable here. It also doesn't seem to be used again in this test. Does it need to be an @?

File: spec/graphql/mutations/stories/delete_story_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        post '/graphql', params: { query: query }
+
+        response = parse_json[:data]
+        expect(response[:deleteStory][:story][:title]).to eq('Deleted')
+        expect(Story.last.title).to_not eq('Deleted')
+      end
+
+      it 'cant find story to delete' do
+        stories = create_list(:story, 5)
```
Comment (jbreit88) [inline spec/graphql/mutations/stories/delete_story_spec.rb:54]: Curious if this needs to be set to a variable since we don't call it later. Could we not just create stories here witout setting it to a variable?

File: spec/graphql/mutations/stories/update_story_spec.rb [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                word
+              }
+            }
+          }
+        GQL
+      end
+
+      it 'updates a story' do
+        @story = create(:story)
+
```
Comment (jbreit88) [inline spec/graphql/mutations/stories/update_story_spec.rb:94]: I don't think this needs to be a global variable, and not sure these factories need to be set to a variable in most cases since we don't call them.

File: spec/graphql/mutations/users/update_user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+          post '/graphql', params: { query: query }
+
+          message = parse_json[:errors].first[:message]
+
+          expect(message).to eq("User does not exist.")
+        end
+
+        it 'does not update with other attributes (sad path)' do
+          @bob = create(:user, id: 1001, username: "Bob", email: "Bobathan@gmail.com")
```
Comment (jbreit88) [inline spec/graphql/mutations/users/update_user_spec.rb]: @ sign un-needed?

**Review by jbreit88 — COMMENTED** [review]
Great job with all the testing. I particularly appreciate the areas where you marked out 'sad path' in the tests. We might even want to add a label of "edge case" for edge case specific tests.

I don't think this would be any reason not to merge the PR seeing as how it is mostly cosmetic, but I saw two patterns in the tests that I wanted to ask y'all about.

1. Some of the variables are assigned as global variables with '@' symbols on them. I am not super familiar with testing GraphQL, so I don't know if that is necessary or not. If it is not necessary I would suggest that we go through and remove them, if they are necessary, my apologies please do carry on (and when you have a moment please tell me why :) )

2. I saw and commented some instances in the test suite where things like factories are set to variables that are never called again. I don't mind them being there, and I I think in some cases they might be able to help the readability of the code, but overall, when we don't call the variable again, I don't know if it is necessary to set a factory to be a variable.

Overall I think this is OK to merge since most of my suggestions are cosmetic, but would be nice to go back and touch those up.

File: spec/graphql/mutations/users/create_user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+module Mutations
+  module Users
+    RSpec.describe CreateUser, type: :request do
+      describe 'create user' do
+        def query
+          <<~GQL
+            mutation {
+              createUser(input:
+                { username: "Lassie", email: "sillyboy@inwell.com" } ) {
+      	      user {
```
Comment (dara-rockwell) [inline spec/graphql/mutations/users/create_user_spec.rb]: ⛏️ Love the use of actual GQL! Generated GQL can be harder to use. I might format the tabbing here so it is apparent which curly brackets go with each other!

File: app/graphql/mutations/stories/update_story.rb [hunk 7]
```diff
@@ -17,6 +17,8 @@ def resolve(attributes)
           story.update(attributes)
 
           {story: story}
+        rescue ActiveRecord::RecordNotFound => _e
+          GraphQL::ExecutionError.new('Story does not exist.')
```
Comment (he-dslieker) [inline app/graphql/mutations/stories/update_story.rb:21]:
```suggestion
        rescue ActiveRecord::RecordNotFound => _e
          GraphQL::ExecutionError.new('Story not found.')
```

**Review by dara-rockwell — APPROVED** [review]

File: spec/graphql/mutations/stories/create_story_spec.rb (hunk 1 above)
Comment (kevingloss) [reply spec/graphql/mutations/stories/create_story_spec.rb:34]: Similar to using a variable here but the syntax for graphql gets a little goofy and with the multiline response it made sense to implement it like this to us.

File: spec/graphql/mutations/stories/delete_story_spec.rb (hunk 2 above)
Comment (kevingloss) [reply spec/graphql/mutations/stories/delete_story_spec.rb:44]: Removed, nice catch.

File: spec/graphql/mutations/stories/delete_story_spec.rb (hunk 3 above)
Comment (kevingloss) [reply spec/graphql/mutations/stories/delete_story_spec.rb:54]: It doesn't need to be a variable, just added extra stories.
